src/Registration/pReg/tests/test_pSIRFReg.py

```diff
@@ -0,0 +1,816 @@
+# Imports
+import os
+import sys
+import time
+
+import numpy as np
+import pSIRFReg
+import pSTIR
+from pUtilities import *
+
+# Paths
+SIRF_PATH = os.environ.get('SIRF_PATH')
+examples_path = SIRF_PATH + '/data/examples/Registration'
+output_prefix = os.getcwd() + '/results/python_'
+
+# Input filenames
+ref_aladin_filename = examples_path + "/test.nii.gz"
+flo_aladin_filename = examples_path + "/test2.nii.gz"
+ref_f3d_filename = examples_path + "/mouseFixed.nii.gz"
+flo_f3d_filename = examples_path + "/mouseMoving.nii.gz"
+parameter_file_aladin = examples_path + "/paramFiles/niftyreg_aladin.par"
+parameter_file_f3d = examples_path + "/paramFiles/niftyreg_f3d.par"
+
+# Output filenames
+save_nifti_image = output_prefix + "save_NiftiImage.nii"
+save_nifti_image_3d = output_prefix + "save_NiftiImage3D.nii"
+save_nifti_image_3d_tensor_not_split = output_prefix + "save_NiftiImage3DTensor_not_split.nii"
+save_nifti_image_3d_tensor_split = output_prefix + "save_NiftiImage3DTensor_split_%s.nii"
+save_nifti_image_3d_deformation_not_split = output_prefix + "save_NiftiImage3DDeformation_not_split.nii"
+save_nifti_image_3d_deformation_split = output_prefix + "save_NiftiImage3DDeformation_split_%s.nii"
+save_nifti_image_3d_displacement_not_split = output_prefix + "save_NiftiImage3DDisplacement_not_split.nii"
+save_nifti_image_3d_displacement_split = output_prefix + "save_NiftiImage3DDisplacement_split_%s.nii"
+aladin_warped = output_prefix + "aladin_warped.nii"
+f3d_warped = output_prefix + "f3d_warped.nii"
+TM_fwrd = output_prefix + "TM_fwrd.txt"
+TM_back = output_prefix + "TM_back.txt"
+aladin_def_fwrd = output_prefix + "aladin_def_fwrd.nii"
+aladin_def_back = output_prefix + "aladin_def_back_%s.nii"
+aladin_disp_fwrd = output_prefix + "aladin_disp_fwrd.nii"
+aladin_disp_back = output_prefix + "aladin_disp_back_%s.nii"
+f3d_def_fwrd = output_prefix + "f3d_disp_fwrd.nii"
+f3d_def_back = output_prefix + "f3d_disp_back_%s.nii"
+f3d_disp_fwrd = output_prefix + "f3d_disp_fwrd.nii"
+f3d_disp_back = output_prefix + "f3d_disp_back_%s.nii"
+
+rigid_resample = output_prefix + "rigid_resample.nii"
+nonrigid_resample_disp = output_prefix + "nonrigid_resample_disp.nii"
+nonrigid_resample_def = output_prefix + "nonrigid_resample_def.nii"
+output_weighted_mean = output_prefix + "weighted_mean.nii"
+output_weighted_mean_def = output_prefix + "weighted_mean_def.nii"
+output_float = output_prefix + "reg_aladin_float.nii"
+
+ref_aladin = pSIRFReg.NiftiImage3D(ref_aladin_filename)
+flo_aladin = pSIRFReg.NiftiImage3D(flo_aladin_filename)
+ref_f3d = pSIRFReg.NiftiImage3D(ref_f3d_filename)
+flo_f3d = pSIRFReg.NiftiImage3D(flo_f3d_filename)
+
+# NiftiImage
+def try_niftiimage():
```

Comment:
I intentionally called them something other than `test_` so that I could choose which tests I wanted to run during the debugging. You're right, I should do this before merging.